lib/Sema/PreCheckExpr.cpp

```diff
@@ -749,10 +749,27 @@ Expr *TypeChecker::resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE,
                                            /*Implicit=*/true);
     }
 
+    auto isInClosureContext = [&](ValueDecl *decl) -> bool {
+      auto *DC = decl->getDeclContext();
+      do {
+        if (dyn_cast_or_null<ClosureExpr>(DC))
```

Comment:
Hey @xedin :) 
Quick question, `dyn_cast_or_null` use here means that DC can be `nullptr`? If it can be, on the first iteration that means we could crash in `DC->getParent()`? Does it makes sense?

Reply:
Ah yes, it should be dyn_cast

Reply:
Fixed, thanks!